### PR TITLE
AP-2783: Rails 7 update

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2783)

This is a placeholder - once it is deployed I will add the rails 7 commits and see if
we can replicate the issues we saw in production

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by colinbruce** [comment]
rebased after circle env-var changes

File: config/application.rb [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
-require "active_storage/engine"
-require "action_controller/railtie"
-require "action_mailer/railtie"
-require "action_mailbox/engine"
-require "action_text/engine"
-require "action_view/railtie"
-require "action_cable/engine"
-# explicitly not loading 'sprockets/railtie'
-require "rails/test_unit/railtie"
+require "rails/all"
```
Comment (jsugarman) [inline config/application.rb:4]: **not a blocker**, but the indvidual component give us more fine grained control i would have thought  - for example are we using action_cable and action_mailbox?

**Review by jsugarman — APPROVED** [review]
👍 LGTM

reminder to remove the last TEMP commit

File: config/application.rb (hunk 1 above)
Comment (colinbruce) [reply config/application.rb:4]: Generally speaking most Rails apps have `rails/all`, see [cccd](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/blob/master/config/application.rb#L3)

We only swicthed to fine-grained to remove sprockets when we switched to (or tried to) WebPacker
Sub gems still required sprockets eventually 😢 

This just reverts to the standard rails import

**Pull request merged**